doc/source/matching_strategies.rst

```diff
@@ -0,0 +1,172 @@
+~~~~~~~~~~~~~~~~~~~
+Matching Algorithms
+~~~~~~~~~~~~~~~~~~~
+
+The core of simply are the different matching algorithms. They decide which bids and asks are
+matched together and at what price. Currently, simply has three matching algorithms: Pay-as-bid,
+Two-Sided Pay-as-clear and BEST matching.
+
+.. _example_scenario:
+
+Example Scenario
+================
+
+The following example scenario is used to illustrate the functioning of the matching mechanisms
+used across the simply matching algorithms. The scenario consists of a basic network of 5 actors
+across 2 clusters. Actors 0 and 4 are in Cluster 0 and Actors 1, 2, and 3 are in Cluster 1. There
+is a +1 grid-fee in-between the Cluster 0 and 1. Matches made between clusters therefore incur an
+additional cost of +1.
+
+Network
+-------
+.. figure:: network.png
+   :scale: 35%
+   :alt: Power Network
+
+   Figure 1: basic network consisting of 5 actors across 2 clusters.
+
+Grid-Free Matrix
+----------------
+
+Grid-fees are calculated for each order using the following grid fee matrix:
+
+.. math::
+
+    \[\left[ {\begin{array}{cc} 0 & 1 \\ 1 & 0 \\ \end{array} } \right]\]
+
+Bids and offers
+---------------
+
+Bids are placed by actors looking to buy energy and offers are placed by actors looking to sell
+energy. For this example, using the network depicted above, 5 orders consisting of 3 bids and 2 asks are
+added into the market.
+
+**The Bids**
+
+Two bids, each for 0.1 energy have been placed by Actor 1 in Cluster 1, one bid is set at the price
+of 10 and another bid at the price of 7. The third bid is placed by Actor 0 in Cluster 0 and is
+for 0.1 energy at the price 10.
+
+**The Offers**
+
+Both offers added into the market are placed by Actor 3 in Cluster 1. Both are for 0.1 energy
+with one set at the price of 6 and another at the price of 4.
+
+Below shows the syntax for inputting the bids and offers:
+
+.. code:: python
+
+    Order = namedtuple("Order", ("type", "time", "actor_id", "cluster", "energy", "price"))
+
+    # Add bids
+    m.accept_order(Order(-1, 0, 1, 1, 0.1, 10))
+    m.accept_order(Order(-1, 0, 1, 1, 0.1, 7))
+    m.accept_order(Order(-1, 0, 0, 0, 0.1, 10))
+
+    # Add asks
+    m.accept_order(Order(1, 0, 3, 1, 0.1, 6))
+    m.accept_order(Order(1, 0, 3, 1, 0.1, 4))
+
+Pay-as-Bid Algorithm
+====================
+
+Using the Pay-as-Bid algorithm (found here: :ref:`market`), buyers are able to place bids in the
+market, alongside
+the
+asks placed by sellers. At each timestamp, bids are sorted in descending order and asks
+are sorted in ascending order. Each offer is then compared with each bid, if the bid is greater
+than the ask, the ask and the bid are matched.
```

Comment:
if the bid price is greater than the ask price, they are matched (partial matches of varying energy quantities are allowed).